Commons/bungee/src/cl/bgmp/bungee/commands/staffchat/StaffChatManager.java

```diff
@@ -0,0 +1,100 @@
+package cl.bgmp.bungee.commands.staffchat;
+
+import cl.bgmp.bungee.ChatConstant;
+import cl.bgmp.bungee.ChatState;
+import cl.bgmp.bungee.CommonsBungee;
+import cl.bgmp.bungee.FlashComponent;
+import cl.bgmp.bungee.Permission;
+import cl.bgmp.bungee.Util;
+import java.util.HashMap;
+import net.md_5.bungee.api.ChatColor;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import net.md_5.bungee.api.event.ChatEvent;
+import net.md_5.bungee.api.event.ServerConnectEvent;
+import net.md_5.bungee.api.plugin.Listener;
+import net.md_5.bungee.event.EventHandler;
+import org.jetbrains.annotations.NotNull;
+
+public class StaffChatManager implements Listener {
+  public static HashMap<String, ChatState> playerChatStates;
+
+  public static HashMap<String, ChatState> getPlayerChatStates() {
+    return playerChatStates;
+  }
+
+  public static void alternatePlayerChat(ProxiedPlayer player) {
+    final ChatState alternation =
+        getPlayerChatStates().get(player.getName()) == ChatState.GLOBAL
+            ? ChatState.STAFF
+            : ChatState.GLOBAL;
+    final ChatConstant message =
+        alternation == ChatState.GLOBAL
+            ? ChatConstant.LOCKED_GLOBAL_CHAT
+            : ChatConstant.LOCKED_STAFF_CHAT;
+
+    getPlayerChatStates().put(player.getName(), alternation);
+    player.sendMessage(
+        new FlashComponent(ChatConstant.STAFF_CHAT_PREFIX.getAsString())
+            .append(message.getAsString())
+            .build());
+  }
+
+  public static void sendStaffChatMsg(@NotNull ProxiedPlayer sender, @NotNull String message) {
+    for (ProxiedPlayer onlinePlayer : CommonsBungee.get().getProxy().getPlayers()) {
+      if (onlinePlayer.hasPermission(Permission.STAFF_CHAT_SEE.getNode())) {
+        onlinePlayer.sendMessage(constructStaffChatMessage(sender, onlinePlayer, message));
+      }
+    }
+  }
+
+  @EventHandler
+  public void onPlayerChat(ChatEvent event) {
+    if (event.isCommand() || !(event.getSender() instanceof ProxiedPlayer)) return;
+
+    final ProxiedPlayer player = (ProxiedPlayer) event.getSender();
+    if (getPlayerChatStates().get(player.getName()) == ChatState.GLOBAL) return;
+
+    event.setCancelled(true);
+
+    for (ProxiedPlayer onlinePlayer : CommonsBungee.get().getProxy().getPlayers()) {
+      if (onlinePlayer.hasPermission(Permission.STAFF_CHAT_SEE.getNode())) {
+        onlinePlayer.sendMessage(
+            constructStaffChatMessage(player, onlinePlayer, event.getMessage()));
+      }
+    }
```

Comment:
this is `sendStaffChatMsg`

Reply:
Indeed! I knew there was somethin odd with having that there lol. https://github.com/ElMedievo-UdeC/Network/pull/31/commits/1353965536c131b4a10f6c032e7ba541584d9540